pkg/apis/hive/v1alpha1/clusterdeployment_types.go

```diff
@@ -69,8 +69,8 @@ type ClusterDeploymentSpec struct {
 	Platform `json:"platform"`
 
 	// PullSecret is the reference to the secret to use when pulling images.
-	// +required
-	PullSecret corev1.LocalObjectReference `json:"pullSecret"`
+	// +optional
+	PullSecret corev1.LocalObjectReference `json:"pullSecret,omitempty"`
```

Comment:
This should be a pointer now that it is optional.

Reply:
Thanks for pointing this out.